lib/gui/modules/drive-scanner.js

```diff
@@ -16,86 +16,140 @@
 
-const availableDrives = Rx.Observable.timer(
-  DRIVE_SCANNER_FIRST_SCAN_DELAY_MS,
-  DRIVE_SCANNER_INTERVAL_MS
-)
+  /**
+   * @summary DriveScanner constructor
+   * @param {Object} [options] - EventEmitter options
+   * @example
+   * var scanner = new DriveScanner()
+   *   .on('error', (error) => { ... })
+   *   .on('drives', (drives) => { ... })
+   */
+  constructor(options) {
+    super(options);
+    this.isRunning = false;
+    this.timer = null;
+    this.scanCount = 0;
+    this.errorCount = 0;
+    this.run = this.run.bind(this);
+  }
 
-/* eslint-enable lodash/prefer-lodash-method */
+  /**
+   * @summary Start the scanner
+   * @public
+   * @example
+   * scanner.start()
+   */
+  start() {
+    debug('start');
+    this.isRunning = true;
+    this.timer = setTimeout(this.run, DRIVE_SCANNER_FIRST_SCAN_DELAY_MS);
+  }
 
-  .flatMap(() => {
-    return Rx.Observable.fromNodeCallback(drivelist.list)();
-  })
+  /**
+   * @summary Kick off a scan & schedule the next run
+   * @private
+   * @example
+   * scanner.run()
+   */
+  run() {
 
-  .map((drives) => {
-    if (settings.get('unsafeMode')) {
-      return drives;
+    if (!this.isRunning) {
+      return;
     }
 
-    return _.reject(drives, {
-      system: true
+    this.scanCount += 1;
+    DriveScanner.scan((error, drives) => {
+      debug('scan', error || drives);
+      if (error) {
+        this.errorCount += 1;
+        this.emit('error', error);
+      } else {
+        this.emit('drives', drives);
+      }
+      if (this.isRunning) {
+        this.timer = setTimeout(this.run, DRIVE_SCANNER_DELAY_MS);
+      }
     });
-  })
-  .pausable(new Rx.Subject());
 
-/*
- * This service emits the following events:
- *
- * - `drives (Object[])`
- * - `error (Error)`
- *
- * For example:
- *
- * ```
- * driveScanner.on('drives', (drives) => {
- *   console.log(drives);
- * });
- *
- * driveScanner.on('error', (error) => {
- *   throw error;
- * });
- * ```
- */
-availableDrives.subscribe((drives) => {
-  emitter.emit('drives', drives);
-}, (error) => {
-  emitter.emit('error', error);
-});
+  }
 
-/**
- * @summary Start scanning drives
- * @function
- * @public
- *
- * @example
- * driveScanner.start();
- */
-emitter.start = () => {
-  availableDrives.resume();
-};
+  /**
+   * @summary Stop scanning for drives
+   * @public
+   * @example
+   * scanner.stop()
+   */
+  stop() {
+    debug('stop');
+    this.isRunning = false;
+    clearTimeout(this.timer);
+  }
 
-/**
- * @summary Stop scanning drives
- * @function
- * @public
- *
- * @example
- * driveScanner.stop();
- */
-emitter.stop = () => {
-  availableDrives.pause();
-};
+  /**
+   * @summary Scan for drives, automatically filtering out system drives
+   * @private
+   * @static
+   * @param  {Function} callback - callback(error, drives)
+   * @example
+   * DriveScanner.scan((error, drives) => {
+   *   ...
+   * });
+   */
+  static scan(callback) {
+    drivelist.list((error, drives) => {
```

Comment:
Should we promisify `drivelist` and use promises here, for consistency with the rest of the Etcher codebase?